docs/guides/dependency-installation.rst

```diff
@@ -1,229 +1,245 @@
+     - * Can't be reliably used to refer to an unpublished, WIP projects.
 
-For illustration, the following example specifies a task that imports its source code from a GitHub repo, and depends on several python modules and a GitLab repo.
+   * - ``GithubImport``
+     - * Well-suited for unpublished, WIP projects.
+       * More robust than ``InlineImport``---it's not prone to pickling errors.
+       * Doesn't inflate the internal workflow definition representation.
+     - * Requires a setuptools-like package manifest (``pyproject.toml``/``setup.cfg``/``setup.py``, etc.)
+       * Requires setting up an access token if the repo isn't public.
+       * Requires committing and pushing code edits before they can be used in a remote workflow.
+       * Doesn't work with other git hostings than GitHub.
+       * Doesn't work with tasks defined in a Jupyter notebook.
 
-.. code-block:: python
+   * - ``GitImportWithAuth``
+     - * All pros of ``GithubImport``.
+       * Is likely to work with other git hostings than GitHub.
+       * Might support more complicated use cases than ``GithubImport``.
+     - * All cons of ``GithubImport``, apart from being GitHub-exclusive.
+       * Verbose. It's easy to make a spelling mistake when specifying its parameters.
 
-    import orquestra.sdk as sdk
 
-    @sdk.task(
-        source_import = sdk.GithubImport(
-            "zapatacomputing/my_source_repo",
-            git_ref = "feat/my-experimental-feature",
-        ),
-        dependency_imports = [
-            sdk.PythonImports("typing_extensions==4.5.0", "pytest~=6.2.5"),
-            sdk.GitImport(
-                repo_url = "https://example.com/some_person/some_repo.git",
-                git_ref = "v1.0.1",
-            )
-        ]
-    )
-    def demo_task():
-        pass
+Good Practice
+=============
 
-How sources and dependencies are treated depends on the runtime.
+The following advices are sorted by an increasing level of complexity, from "hello world" to "complicated project within a niche use case".
+See also `Imports In Detail`_ for more info about each one.
 
-The examples below omit ``dependency_imports`` for clarity as the usage of an importer is identical regardless of whether it is importing a source or a dependency.
+#. The defaults: ``InlineImport`` as the ``source_import`` and empty ``dependency_imports``.
 
-``GithubImport``
-----------------
+   .. literalinclude:: ../examples/tests/test_dependency_installation.py
+       :start-after: def good_practice_defaults():
+       :end-before: </snippet>
+       :language: python
+       :dedent: 8
 
-The ``GithubImport`` importer is the preferred option where the source or dependency is stored in a GitHub repo. The following minimal example shows a source import from a (fictional) repo located at ``https://github.com/zapatacomputing/my_source_repo``.
 
-.. code-block:: python
+#. ``PythonImports`` set as ``dependency_imports=[...]`` if your task depends on third-party libraries.
```

Comment:
Suggestion: I'd be tempted to lead with the use case in the header, essentially following the first header that starts with "The defaults:...". This would break up the sentence making it easier to parse, and allow skimming down the list for a relevant example without having to read to the end of each heading:

```suggestion
#. If your task depends on third-party libraries: ``PythonImports`` set as ``dependency_imports=[...]``.
```